composer.json: Use current SPDX license identifier

dontub opened the pull request:
Overview
----------------------------------------
`APL-3.0` is a deprecated license identifier and should be replaced by the more specific `AGPL-3.0-only`. https://spdx.org/licenses/

Before
----------------------------------------
A deprecated SPDX license identifier is used in `composer.json`.

After
----------------------------------------
A current SPDX license identifier is used in `composer.json`.

seamuslee001 commented:
@JoeMurray do you have any thoughts on this?

JoeMurray commented:
I did some research. OSI, GNU and FSF do not define two licenses like SPDX does, only one. OSI mentioned SPDX tag of AFPL-3.0-only , other two do not. Our license text includes the template AGPL-3.0 text that includes the phrase 'or later' so I believe our SPDX tag should be AGPL-3.0-or-later not AGPL-3.0-only . However, I am not sure that GNU or FSF would recognize AGPL-3.0-or-later as a license. We want to be compliant with the organization that stewards license, so I have emailed them to see if they recognize SPDX tag AGPL-3.0-or-later, and also what the text in our repo means for which SPDX label we should use.

JoeMurray commented:
@dontub do you seem to be with Systopia. I take it your interest is in CiviCRM having an SPDX license identifier rather than than caring about 3.0-only vs 3.0-or-later. Can you provide a link that shows FSF or GNU recognizing officially SPDX license identifiers? I'm not confident that GNU has the staffing to respond to an email like from me.

JoeMurray reviewed (changes requested):
Please change to AGPL-3.0-or-later since that is what our license has said since it was adopted about 15 years ago. That label is approved by both FSF and OSI as per https://spdx.org/licenses/ .

Commit:
composer.json: Use SPDX license identifier `AGPL-3.0-or-later`

`APL-3.0` is a deprecated license identifier and should be replaced by
the more specific `AGPL-3.0-or-later`. https://spdx.org/licenses/

dontub commented:
Changed it to `AGPL-3.0-or-later`.

JoeMurray commented:
FSF got back to me and I would say that the gist of their response confirmed we should use AGPL-3.0-or-later .

colemanw commented:
@JoeMurray why am I not surprised it wasn't a short simple answer?
